src/connectors/infinite-hits/connectInfiniteHits.js

```diff
@@ -96,29 +119,45 @@ export default function connectInfiniteHits(renderFn, unmountFn) {
       },
 
       render({ results, state, instantSearchInstance }) {
-        if (state.page === 0) {
+        // Reset cache and received pages if anything changes in the
+        // search state, except for the page.
+        //
+        // We're doing this to "reset" the widget if a refinement or the
+        // query changes between renders, but we want to keep it as is
+        // if we only change pages.
+        const { page, ...stateWithoutPage } = state;
+        const currentState = JSON.stringify(stateWithoutPage);
```

Comment:
Is a comparaison with a stringified object safe in that case? Are we sure that the helper state always orders its state keys in the same order?

Since we use `lodash.isEqual` in the codebase, it might be better to rely on their implementation. According to [this benchmark](https://www.measurethat.net/Benchmarks/Show/1854/0/lodash-isequal-test), there are no important performance concerns to have.

WDYT?

Reply:
From what I saw the state is an instance of `SearchParameters` and it's always constructed the same way, so it the keys are always in the same order.

For lodash, I though we were in the process of removing it from our dependencies?

Reply:
We're in the process of removing Lodash in the JavaScript helper, not in InstantSearch.js (not even sure we plan this).

Even if we were to remove Lodash, I think it's safer to use it here since we already often rely on their `isEqual` util function. We can still later think of creating our own `isEqual` util and replacing the import.

Would that make sense?

Reply:
> We're in the process of removing Lodash in the JavaScript helper, not in InstantSearch.js (not even sure we plan this).

Well it seems logical to me to remove it from InstantSearch.js after the helper :wink:

> Even if we were to remove Lodash, I think it's safer to use it here since we already often rely on their isEqual util function. We can still later think of creating our own isEqual util and replacing the import.

I'm ok with that :slightly_smiling_face: